cra-flutter/package.json

```diff
@@ -0,0 +1,52 @@
+{
+  "name": "cra-flutter",
+  "version": "0.0.0",
+  "scripts": {
+    "prestart": "rimraf node_modules/.cache/",
+    "start": "react-scripts start",
+    "prebuild": "pushd flutter && flutter clean && flutter build web --output=../public/flutter && popd",
```

Comment:
This is building in production mode, I don't understand the error with canvaskit then :/

Reply:
I saw the error with the canvaskit using production version, but it is currently gone after I refactored code from
```js
_flutter.loader.loadEntrypoint({
   entrypointUrl: src,
   onEntrypointLoaded: async (engineInitializer: any) => {
    let appRunner = await engineInitializer.initializeEngine({
      hostElement: ref.current,
      assetBase: assetBase,
    });
    await appRunner.runApp();
  }
});
```
to
```js
const initFlutterApp = async () => {
  const engineInitializer = await new Promise<any>((resolve) => {
   _flutter.loader.loadEntrypoint({
      entrypointUrl: src,
      onEntrypointLoaded: resolve,
    })
  })

  const appRunner = await engineInitializer?.initializeEngine({
    hostElement: ref.current,
    assetBase: assetBase,
  })
  await appRunner?.runApp()
}
initFlutterApp()
```

Reply:
Now that I notice, your `initFlutterApp` is `async` but when you call it, do you `await initFlutterApp()`?

Not awaiting an async function might introduce race conditions in the startup process (maybe?)

Reply:
In React, we can't make a function we pass into `useEffect` async, so that is the only way.  
And this second version is actually the one that doesn't have the canvaskit issue

Reply:
Is there any way of preventing the multiple calls to the flutter initialization? Something like a check to see if the widget is not mounted or similar?

(I have to admit I was surprised by angular not re-initializing the flutter app multiple times, but of course it's a very simple app, this is eventually a problem if your app has re-renderings)

((Question: is this version still having trouble with re-rendering? :P))

Reply:
> Question: is this version still having trouble with re-rendering? :P

No. Those were my findings during the embedding journey, but the current version is the most stable -- I didn't catch any errors so far. 

(except when I tried to use some community plugins just to check how it will work) 

I even checked the existing component from the Angular app -- added it to the Storybook, so re-rendering will be happened naturally when you navigate between stories :) If you don't share `engineInitializer` between re-renders everything works fine, and also if you're not trying to navigate between tabs really fast (or forcing component reload on purpose using built-in storybook feature). 

But in the React app re-rendering was much faster because it was made programmatically due to the strict mode, as I wrote before. 

> Is there any way of preventing the multiple calls to the flutter initialization? Something like a check to see if the widget is not mounted or similar?

Yes, that was one of the optimizations that I made -- added `isRendered` flag

Reply:
`but you can't fully rely on this flag because of the many async processes that we have during the initialization`

Yes, this should be contrlled in a much tighter way from Flutter, but that's an architectural problem of Flutter Web, and not the embedding. Thanks for getting this to stable!

I'll try to build this and have it deployed somewhere (or maybe we can put it in github pages of this repo?) so we can announce it on twitter and people are aware of it?

I think people are going to have fun checking this out!

Reply:
let me check the github pages possibility 

Meanwhile, I think this PR can be merged? We can continue discussion in the thread or in the Discord

Reply:
> Meanwhile, I think this PR can be merged? 

I think so too. Ship it!

Reply:
Published: https://p-mazhnik.github.io/flutter-embedding

Found that TextField can't loose focus when clicking outside of the Flutter component (reproducible in the Angular sample as well), but I believe this is known?

Reply:
> So, if you are going to reference a basic React example somewhere, perhaps you could also mention the React Native one?

Absolutely! I'll send a PR to the flutter/samples repo linking to your code. Are they in separate repos/directories? I think it'd be best if they had a README file in the location I link them to, so people can immediately read the docs.

Reply:
Same repo, separate directories 
- React: https://github.com/p-mazhnik/flutter-embedding/tree/main/cra-flutter
  - Deploy: https://p-mazhnik.github.io/flutter-embedding
- React-Native (Expo): https://github.com/p-mazhnik/flutter-embedding/tree/main/expo-flutter 

Directories also have a short description in the root README file.

Reply:
Perhaps I should deploy React-Native Web part somewhere as well